qemu/tests/qemu_img_map_unaligned_image.py

```diff
@@ -0,0 +1,51 @@
+import logging
+import random
+import json
+import string
+
+from avocado.utils import process
+from virttest import data_dir
+from virttest.qemu_storage import QemuImg
+
+
+def run(test, params, env):
+    """
+    qemu-img map an unaligned image.
+
+    1.create a raw file using truncate
+    2.write data into the raw file
+    3.qeme-img map to verify map function works as expected
+
+    :param test: Qemu test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    def _generate_random_string(max_len=19):
+        """Generate random alphabets string in the range of [1, max_len]."""
+        random_str = ''.join(random.choice(
+            string.ascii_lowercase) for _ in range(random.randint(0, max_len)))
+        return random_str, len(random_str)
+
+    def _verify_qemu_img_map(output, str_len):
+        """Verify qemu-img map's output."""
+        # str_len + 1 is expected qeme-img map output for length
+        expected = {"start": 0, "length": str_len + 1, "depth": 0,
+                    "zero": False, "data": True, "offset": 0}
+        res = json.loads(output)[0]
+        if res != expected:
+            test.fail("qemu-img map on an unaligned image is not correct.")
```

Comment:
'The dumped metedata info of the unaligned image is not correct' would be better, thanks.